src/main/kotlin/no/skatteetaten/aurora/boober/model/AuroraConfig.kt

```diff
@@ -112,7 +112,13 @@ data class AuroraConfig(
             return Pair(newFile, this.copy(files = files))
         }
 
+        // We are adding a new file
+        if (previousVersion == null) {
```

Comment:
Jeg mener at updateFile kalles både når det legges inn en ny fil i auroraconfig og når det endres en eksisterende fil.  Når previousVersion er null, så indikerer det at det forventes at det er en ny fil og at det ikke finnes en fil fra før - altså forventes det at currentFile.version som det testes for lenger ned er null. Derfor blir det feil å kaste PreconditionFailureException bare fordi previousVersion er null.

Jeg tror denne feilen skal kastes kun når currentFile.version != previousVersion.  Feilen detekteres jo riktig fra før, men meldingen er feil, så jeg ville plassert denne Innenfor samme if-struktur som der det kastes AuroraVersioningException. F.eks:

```
        if (currentFile.version != previousVersion) {
            if (previousVersion == null) {
                throw PreconditionFailureException("The fileName=$name already exists in this AuroraConfig.")
            } else {
                throw AuroraVersioningException(this, currentFile, previousVersion)
            }
        }
```

Reply:
Jeg tror du må lese koden litt lengre oppe i metoden som ikke er endret. Der tar man hånd om hva som skjer hvis currentFile ikke finnes.

Reply:
Vi må i så fall avklare korrekt oppførsel for caset der previousVersion har en non-null verdi og filen ikke finnes.

Reply:
Den er jeg med på. Basically man tror at det skal finnes en fil med en versjon men den finnes ikke.